src/parser/matches/arg_matches.rs

```diff
@@ -1484,6 +1766,14 @@ fn unwrap_string(value: &AnyValue) -> &str {
     }
 }
 
+fn unwrap_downcast_ref<T: Any + Clone + Send + Sync + 'static>(value: &AnyValue) -> &T {
+    value.downcast_ref().expect(INTERNAL_ERROR_MSG)
+}
+
+fn unwrap_downcast_into<T: Any + Clone + Send + Sync + 'static>(value: AnyValue) -> T {
+    value.downcast_into().expect(INTERNAL_ERROR_MSG)
```

Comment:
These need `#[track_caller]` which makes me wonder if breaking these out into functions is worth it.